### PR TITLE
Add fake trajectory to PixelTracks

### PR DESCRIPTION
**Pull request by JanFSchulte**
Add a fake trajectory and dummy chi2 values to the PixelTracks class so that pixel tracks can be used with classes such as TrackListMerger. Needed for mitigation of the effect of non-functioning modules in the new pixel detector

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by fwyzard** [comment]
please comment or sign and merge ASAP

**Comment by slava77** [comment]
sorry for the delay. I should provide feedback or sign later today.

**Comment by fwyzard** [comment]
thanks (I just discovered we need this PR)

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/18816	f92a8df
- code change sets default trajectory params and chi2 per hit in PixelTrackProducer. This is used mainly in HLT, but also in RecoHI/HiTracking. The cost for HI reco would be some small increase in memory (o(1MB)/thread corresponding to ~100 bytes/track and at worst a few K tracks).
- jenkins tests pass and comparisons with baseline show no differences

File: RecoPixelVertexing/PixelTrackFitting/plugins/PixelTrackProducer.cc [hunk 1]
```diff
@@ -92,6 +93,14 @@ void PixelTrackProducer::store(edm::Event& ev, const TracksWithTTRHs& tracksWith
     unsigned int nHits = tracks->at(k).numberOfValidHits();
     theTrackExtra.setHits(hitCollProd, cc, nHits);
     cc +=nHits;
+    reco::TrackExtra::TrajParams trajParams;
```
Comment (davidlange6) [inline RecoPixelVertexing/PixelTrackFitting/plugins/PixelTrackProducer.cc]: Hi @JanFSchulte - please simplify this to 
reco::TrackExtra::TrajParams trajParams(nHits,LocalTrajectoryParameters(v,1.));
reco::TrackExtra::Chi2sFive chi2s(nHits,0);
theTrackExtra.setTrajParams(std::move(trajParams),std::move(chi2s));

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/18816	438c6fb
- updated since last signoff following  comments from @davidlange6 to initialize vectors in a constructor rather than in a loop
- jenkins tests pass; no differences

**Comment by davidlange6** [comment]
+1

**Pull request merged**